app/reports_generation/scripts/csv_aggregator.py

```diff
@@ -33,39 +33,45 @@ def __validate_config(config: dict):
 def __create_header(config) -> List[str]:
     header = ['Action']
     [header.append(run['runName']) for run in config['runs']]
+    # Append 'App-specific' header
+    header.append('App-specific')
 
     return header
 
 
-def __validate_count_of_actions(tests_results: List[dict]):
+def __validate_count_of_actions(tests_results: List[ResultsCSV]):
     if any(len(tests_results[0].actions) != len(actions_count.actions) for actions_count in tests_results):
         for file in tests_results:
             print(f'Result file {file.absolute_file_path} has {len(file.actions)} actions\n')
         raise SystemExit('Incorrect number of actions. '
                          'The number of actions should be the same for each results.csv.')
 
 
-def __get_tests_results(config: dict) -> List[dict]:
+def __get_tests_results(config: dict) -> List[ResultsCSV]:
     results_files_list = []
     column_name = config['column_name']
     for run in config['runs']:
         value_by_action = {}
         absolute_file_path = resolve_path(run['fullPath']) / RESULTS_CSV_FILE_NAME
         with absolute_file_path.open(mode='r') as fs:
             for row in csv.DictReader(fs):
-                value_by_action[row['Label']] = row[column_name]
+                value_by_action[row['Label']] = {column_name:row[column_name], 'App-specific': row['App specific']}
```

Comment:
Since we already have class ResultsCSV I suggest creating one more class Action with two fields: value and is_app_specific and use it instead of a dictionary inside ResultsCSV class in accordance with OOP

Reply:
or just change `{column_name:row[column_name]` to `{'value':row[column_name]`

Reply:
Already discussed this moment.